packages/buttons/src/styled/StyledButton.ts

```diff
@@ -49,88 +45,104 @@ export const getHeight = (props: IButtonProps & ThemeProps<DefaultTheme>) => {
  * 3. shifting :focus-visible from LVHFA order to preserve `color` on hover
  * 4. set default outline-color for smooth transition without artifacts
  */
-const colorStyles = (props: IButtonProps & ThemeProps<DefaultTheme>) => {
+const colorStyles = ({
+  theme,
+  isLink,
+  isBasic,
+  isDanger,
+  isNeutral,
+  isPrimary,
+  focusInset
+}: IButtonProps & ThemeProps<DefaultTheme>) => {
   let retVal;
-  let hue;
-
-  if (props.disabled || (props.isNeutral && props.isPrimary && !props.isDanger)) {
-    hue = 'neutralHue';
-  } else if (props.isDanger) {
-    hue = 'dangerHue';
-  } else {
-    hue = 'primaryHue';
-  }
+  const disabledBackgroundColor = getColor({ theme, variable: 'background.disabled' });
+  const disabledForegroundColor = getColor({ theme, variable: 'foreground.disabled' });
+  const offset100 = { dark: { offset: -100 }, light: { offset: +100 } };
+  const offset200 = { dark: { offset: -200 }, light: { offset: +200 } };
+
+  if (isLink) {
+    /*
+     * Anchor / link button styling
+     */
+    const options = { theme, variable: `foreground.${isDanger ? 'danger' : 'primary'}` };
```

Comment:
While elegant, this pattern might complicate searching for variable names. For ex, searching for all instances of `foreground.primary`.

```suggestion
    const options = { theme, variable: isDanger ? 'foreground.isDanger' : 'foreground.primary' };
```

Reply:
I think I agree with this as well... we'll also probably thank ourselves by helping the vars be searchable.